src/metabase/driver/presto.clj

```diff
@@ -0,0 +1,340 @@
+(ns metabase.driver.presto
+  (:require [clojure.set :as set]
+            [clojure.string :as str]
+            [clj-http.client :as http]
+            (honeysql [core :as hsql]
+                      [helpers :as h])
+            [metabase.config :as config]
+            [metabase.driver :as driver]
+            [metabase.driver.generic-sql :as sql]
+            (metabase.models [field :as field]
+                             [table :as table])
+            [metabase.sync-database.analyze :as analyze]
+            [metabase.query-processor.util :as qputil]
+            [metabase.util :as u]
+            [metabase.util.honeysql-extensions :as hx])
+  (:import (java.util Date)
+           (metabase.query_processor.interface DateTimeValue Value)))
+
+
+;;; Presto API helpers
+
+(defn- details->uri
+  [{:keys [ssl host port]} path]
+  (str (if ssl "https" "http") "://" host ":" port path))
+
+(defn- details->headers [{:keys [user catalog report-timezone]}]
+  (merge {"X-Presto-Source" "metabase"
+          "X-Presto-User"   user}
+         (when-not (nil? catalog)
+           {"X-Presto-Catalog" catalog})
+         (when-not (nil? report-timezone)
+           {"X-Presto-Time-Zone" report-timezone})))
+
+(defn- parse-time-str [s]
+  (.parse (java.text.SimpleDateFormat. "HH:mm:ss.SSS") s))
+
+(defn- parse-time-with-tz-str [s]
+  (.parse (java.text.SimpleDateFormat. "HH:mm:ss.SSS z") s))
+
+(defn- parse-timestamp-str [s]
+  (.parse (java.text.SimpleDateFormat. "yyyy-MM-dd HH:mm:ss.SSS") s))
+
+(defn- parse-timestamp-with-tz-str [s]
```

Comment:
These didn't work anyway after all, I'm changing it to use util/parse-date instead, so similar to this.